BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2089,6 +2174,39 @@ void AAMGuidance()
                     aamTarget = MissileGuidance.GetAPNTarget(TargetPosition, TargetVelocity, TargetAcceleration, vessel, pronavGain, out timeToImpact);
                 else if (GuidanceMode == GuidanceModes.PN) // Pro-Nav
                     aamTarget = MissileGuidance.GetPNTarget(TargetPosition, TargetVelocity, vessel, pronavGain, out timeToImpact);
+                else if (GuidanceMode == GuidanceModes.AAMLoft)
+                {
+                    float targetAlt = FlightGlobals.getAltitudeAtPos(TargetPosition);
+
+                    if (TimeToImpact == float.PositiveInfinity)
+                    {
+                        if ((vessel.altitude >= LoftMinAltitude) && ((vessel.altitude - targetAlt <= LoftAltitudeAdvMax) || Vector3.Distance(TargetPosition, vessel.transform.position) > LoftRangeOverride)) loftState = 0;
```

Comment:
Instead of `Vector3.Distance(TargetPosition, vessel.transform.position) > LoftRangeOverride`, prefer using `(TargetPosition - vessel.transform.position).sqrMagnitude > LoftRangeOverride * LoftRangeOverride` to avoid the `sqrt` if the distance isn't actually needed.

Reply:
Fixed